public/app/features/dashboard/containers/SoloPanelPage.tsx

```diff
@@ -57,8 +57,8 @@ export class SoloPanelPage extends Component<Props, State> {
       return;
     }
 
-    // we just got the dashboard!
-    if (!prevProps.dashboard) {
+    // we just got a new dashboard
+    if (!prevProps.dashboard || prevProps.dashboard.uid !== dashboard.uid) {
```

Comment:
this can never happen, if you navigate to another dashboard there will be an unmount new SoloPage